tests/basic/nomad_test.go

```diff
@@ -413,3 +466,798 @@ var _ = Describe("{RunFioJobWithSnapshotOperationsOnNomad}", func() {
+			_, err := task.DoRetryWithTimeout(func() (interface{}, bool, error) {
+				err := client.DeleteVolume(volumeID)
+				if err != nil {
+					log.Errorf("Retry error: failed to delete volume %v: %v", volumeID, err)
+					return nil, true, err
+				}
+				return nil, false, nil
+			}, 2*time.Minute, 20*time.Second)
+
+			if err != nil {
+				log.Errorf("Final error: failed to delete volume %v: %v", volumeID, err)
+			} else {
+				log.Infof("Successfully deleted volume %v", volumeID)
+			}
+		}
+	})
+})
+
+var _ = Describe("{RebootNodeWhileAppsAreRunning}", func() {
+	var client *nomad.NomadClient
+	var err error
+	var volumeID, pluginID string
+	var volumeIDs []string
+	var jobIDs []string
+	var directories = []string{"job1", "job2", "job3"}
+
+	BeforeEach(func() {
+		StartTorpedoTest("RebootNodeWhileAppsAreRunning", "Runs Multiple FIO Jobs on Nomad Cluster and reboots a node", nil, 0)
+		client, err = nomad.NewNomadClient()
+		log.FailOnError(err, "Failed to get Nomad Client")
+
+		volumeID = fmt.Sprintf("fio-test-volume-%v", time.Now().Unix())
+		pluginID = "portworx"
+		capacityMin := int64(50 * 1024 * 1024 * 1024)
+		capacityMax := capacityMin
+		accessMode := "multi-node-multi-writer"
+		attachmentMode := "file-system"
+		fsType := "ext4"
+		mountFlags := []string{}
+		parameters := map[string]string{
+			"repl":        "2",
+			"io_profile":  "db_remote",
+			"priority_io": "high",
+		}
+
+		err = client.CreateVolume(volumeID, pluginID, capacityMin, capacityMax, accessMode, attachmentMode, fsType, mountFlags, parameters)
+		log.FailOnError(err, "Failed to create shared volume")
+		log.Infof("Shared volume %v created successfully", volumeID)
+		volumeIDs = append(volumeIDs, volumeID)
+		for _, directory := range directories {
+			jobID := fmt.Sprintf("fio-%s", directory)
+			jobIDs = append(jobIDs, jobID)
+			fioJob := client.CreateFioJobSpec(volumeID, jobID, directory)
+			err = client.CreateJob(fioJob)
+			log.FailOnError(err, "Failed to create Fio job for directory "+directory)
+		}
+		log.Infof("Sleeping for 10 seconds to allow allocations to start")
+		time.Sleep(10 * time.Second)
+
+		// Ensure all jobs are running
+		for _, jobID := range jobIDs {
+			_, err = task.DoRetryWithTimeout(func() (interface{}, bool, error) {
+				status, err := client.CheckJobStatus(jobID)
+				if err != nil || status != "running" {
+					return nil, true, fmt.Errorf("job %s is not running yet", jobID)
+				}
+				return nil, false, nil
+			}, 5*time.Minute, 10*time.Second)
+			log.FailOnError(err, "Failed to wait for job to be in running state")
+		}
+	})
+
+	It("Should reboot a node, wait for it to come back, and validate FIO jobs", func() {
+		nodes, err := client.ListNodes()
+		log.FailOnError(err, "Failed to list nodes")
+		if len(nodes) > 0 {
+			node := nodes[0]
+			log.Infof("Rebooting node: %v", node.Name)
+			rebootCmd := `reboot`
+			_, err = client.ExecCommandOnNodeSSH(node.ID, rebootCmd)
+			if err == nil {
```

Comment:
if check not required here

Reply:
No the check is if there is no err then we fail - as the command we are executing is reboot of node - so we have to do this negative check

Reply:
We are expecting an error here